### PR TITLE
[gitlab] Update JOBOWNERS and Slack channels after Agent Platform split

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Updates mentions of Agent Platform in `JOBOWNERS` and tooling to the new teams.

### Motivation

Send notifications to the correct channels.

### Additional Notes

n/a

### Possible Drawbacks / Trade-offs

n/a

### Describe how to test/QA your changes

n/a

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM for Agent Developer Experience !

File: .gitlab/JOBOWNERS [hunk 1]
```diff
@@ -28,6 +28,30 @@ build_serverless*                    @DataDog/serverless
...
+dogstatsd_suse*                      @DataDog/agent-build-and-releases
+agentless_scanner_suse*              @DataDog/agent-build-and-releases
+agent_dmg*                           @DataDog/agent-build-and-releases
+windows_msi*                         @DataDog/windows-agent
+
+# Testing package deploy
+deploy_deb_testing*                  @DataDog/agent-build-and-releases
+deploy_rpm_testing*                  @DataDog/agent-build-and-releases
+deploy_suse_rpm_testing*             @DataDog/agent-build-and-releases
+deploy_windows_testing*              @DataDog/agent-build-and-releases
```
Comment (chouquette) [inline .gitlab/JOBOWNERS:53]: Just to double check, doesn't this belong to `windows-agent`?

**Review by chouquette — APPROVED** [review]
Open question aside, LGTM

**Review by bkabrda — APPROVED** [review]

File: .gitlab/JOBOWNERS (hunk 1 above)
Comment (KSerrania) [reply .gitlab/JOBOWNERS:53]: This job doesn't really do anything Windows-specific, it only pushes Windows artifacts to the testing repos. I see it as the testing equivalent of the release tooling for our prod repos.

**Pull request merged**